File: backend/secondbrain/src/main/java/uknowklp/secondbrain/api/note/dto/NoteSearchResult.java
package uknowklp.secondbrain.api.note.dto;

import java.time.LocalDateTime;

import lombok.Builder;
import uknowklp.secondbrain.api.note.domain.NoteDocument;

// 검색 결과 DTO
@Builder
public record NoteSearchResult(
	Long id,
	String title,
	String content,
	Long userId,
	LocalDateTime createdAt,
	LocalDateTime updatedAt,
	Integer remindCount
) {
	// NoteDocument를 NoteSearchResult로 변환
	public static NoteSearchResult from(NoteDocument document) {
		return NoteSearchResult.builder()
			.id(document.getId())
			.title(document.getTitle())
			.content(document.getContent())
			.userId(document.getUserId())
			.createdAt(document.getCreatedAt())
			.updatedAt(document.getUpdatedAt())
			.remindCount(document.getRemindCount())
			.build();
	}
}

File: mobile_watch/mobile/src/main/java/com/example/secondbrain/data/model/NoteSearchResult.kt
package com.example.secondbrain.data.model

import com.google.gson.annotations.SerializedName

// 검색 결과 개별 노트 데이터 모델
data class NoteSearchResult(
    @SerializedName("id")
    val id: Long,

    @SerializedName("title")
    val title: String,

    @SerializedName("content")
    val content: String,

    @SerializedName("userId")
    val userId: Long,

    @SerializedName("createdAt")
    val createdAt: String,

    @SerializedName("updatedAt")
    val updatedAt: String,

    @SerializedName("remindCount")
    val remindCount: Int?
)
